NosTests/Service/DatabaseCleanerTests.swift

```diff
@@ -20,10 +20,10 @@ final class DatabaseCleanerTests: CoreDataTestCase {
     
     // MARK: - EventReferences
     
-    func test_cleanup_deletesCorrectEventReferences() async throws {
+    @MainActor func test_cleanup_deletesCorrectEventReferences() async throws {
```

Comment:
Because `testContext` from the superclass is actually the `viewContext` which is bound to the main thread. When you mark a test function `async` then it sometimes gets executed on a background thread, which sometimes Core Data doesn't like and it will cause a test failure.

I tried adding @MainActor to the whole class but got some warning about actor conformance being different from the superclass or something, so I just added the attribute to each function instead.

Reply:
Good to know. I assumed the unit tests always ran in the "main" thread given that there is no limits when testing. But it makes sense that they run in the thread they want.